spec/rspec/mocks/verifying_doubles/class_double_with_class_loaded_spec.rb

```diff
@@ -28,6 +28,16 @@ module Mocks
         prevents { expect(o).to receive(:defined_instance_method) }
       end
 
+      it "suggests using `instance_double` when an instance method is stubbed" do
+        o = class_double("LoadedClass")
+        prevents(/Perhaps you meant to use `instance_double`/) { allow(o).to receive(:defined_instance_method) }
+      end
+
+      it "doesn't suggest `instance_double` when a non-instance method is stubbed'" do
+        o = class_double("LoadedClass")
+        prevents(/(?!Perhaps you meant to use `instance_double`)/) { allow(o).to receive(:undefined_instance_method) }
```

Comment:
This doesn't actually work properly...if you break the implementation (e.g. by changing the conditional to `if true`) this still passes.  I haven't used regex look aheads much so I'm not sure why it's failing, but I think there's a better way to do this:

``` ruby
WRONG_DOUBLE_TYPE_MSG = "Perhaps you meant to use `class_double`"

it "suggests using `class_double` when a class method is stubbed" do
  o = instance_double("LoadedClass")
  prevents(a_string_including(WRONG_DOUBLE_TYPE_MSG)) { allow(o).to receive(:defined_class_method) }
end

it "doesn't suggest `class_double` when a non-class method is stubbed" do
  o = instance_double("LoadedClass")
  prevents(a_string_excluding(WRONG_DOUBLE_TYPE_MSG)) { allow(o).to receive(:undefined_class_method) }
end
```

* This uses `a_string_excluding`, which is defined as the negation of `include` [here](https://github.com/rspec/rspec-mocks/blob/f190e66066108547bff5b833e936ab838d53ffc4/spec/spec_helper.rb#L116)
* `prevents` passes the arg to `raise_error`, which like all RSpec matchers where it makes sense to do so, supports receiving a matcher as an arg.
* I also recommend extracting the message snippet into a constant; that way, the positive/negative cases are kept in sync if the wording ever changes.  In contrast, if the wording was to change, the negative case could just pass because it doesn't match the exact wording even if that part of the message was wrongly included.